nixos/modules/tasks/filesystems.nix

```
@@ -140,7 +140,10 @@ let
         else if config.fsType == "reiserfs" then "-q"
         else null;
     in {
-      options = mkIf config.autoResize [ "x-nixos.autoresize" ];
+      options = mkMerge [
+        (mkIf config.autoResize [ "x-nixos.autoresize" ])
+        (mkIf (utils.fsNeededForBoot config) [ "x-initrd.mount" ])
+      ];
```

```suggestion
      options = [
        (mkIf config.autoResize "x-nixos.autoresize")
        (mkIf (utils.fsNeededForBoot config) "x-initrd.mount")
      ];
```
I just wanted to test my module knowledge: does this work, too? I think so.

---

Looks like no:

```
error: A definition for option `nodes.machine.fileSystems."/mnt".options' is not of type `non-empty (list of string (with check: non-empty))'. Definition values:
       - In `/nix/store/hncd2hanr43scbs0rinn353jipr55k8n-source/nixos/modules/virtualisation/qemu-vm.nix': [ ]
(use '--show-trace' to show detailed location information)
```

---

Oh I see. It does *technically* work as far as the module system is concerned. The issue is that ordinarily the default for this option is `["defaults"]`. But by using `options = [ (mkIf ...) ]` instead of `options = mkMerge [ (mkIf ...) ]`, we are unconditionally setting the option to a list, meaning the default of `["defaults"]` *never* applies, and we end up with an empty list if both `mkIf` conditions are false. But this option is declared such that it can't be an empty list (presumably because with no options we want `"defaults"`)